src/user/models/user/user.controller.ts

```diff
@@ -42,6 +44,14 @@ export class UserController {
     return this.userService.updateUser(jwt.id, newUser);
   }
 
+  @Post('change')
+  @ApiBearerAuth()
+  @UseGuards(AuthGuard(), new RoleGuard(UserRole.USER))
+  async changeUser(@GetJwt() jwt: JwtPayload, @Body() changeUser: LinkedUserDto): Promise<{ accessToken: string }> {
+    const checkedUser = await this.userService.changeUser(jwt.id, changeUser);
+    return this.authService.signIn(checkedUser);
```

Comment:
I would directly use the `AuthService.generateToken` method here.

Reply:
move `changeUser` to `AuthService`